### PR TITLE
[#2] import all tenancy tribunal cases into postgres

### PR DESCRIPTION
**Pull request by mcqueen1987**

### CODE REVIEW

**Comment by mcqueen1987** [comment]
@williamparry please help review

File: getDataFile/generic/url.js [hunk 1]
```diff
@@ -3,27 +3,28 @@ const download = require("download");
...
 
 if (require.main === module) {
-	try {
-		throw new Error("Cannot be run individually.")
-		//run();
-	} catch (ex) {
-		console.log(ex);
-	}
+    try {
+        throw new Error("cannot be run individually.")
```
Comment (williamparry) [inline getDataFile/generic/url.js]: Why this change?

File: getDataFile/index.js [hunk 2]
```diff
@@ -1,48 +1,39 @@
...
-		try {
-			retData = JSON.stringify(JSON.parse(datasource));
-		} catch (ex) {
-			throw ex;
-		}
-	}
-
-	return retData;
-
+module.exports = async (pgPool, pgPromise, dataSource, dataLocation) => {
```
Comment (williamparry) [inline getDataFile/index.js]: This file should not know about a database connection. It is just there to get the data, not put it in the DB

File: getTenancyTribunal.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * get tenancy tribunal data and save to postgres
+ *
+ * @param pgPool
+ * @param pgPromise
+ * @param datasource
+ * @param datalocation
+ * @returns {Promise<void>}
+ */
+const run = async (pgPool, pgPromise, datasource, datalocation) => {
+    await getDataFile(pgPool, pgPromise, datasource, datalocation);
```
Comment (williamparry) [inline getTenancyTribunal.js]: This is where you put the DB logic.

File: getDataFile/generic/url.js (hunk 1 above)
Comment (mcqueen1987) [reply getDataFile/generic/url.js]: 1, add code 'if (!common.isJsonString(jsonData)) ......' is because some of the data from server may not the valide Json fromate, we need to theck and log it. otherwise throw an error when we call JSON.parse function.
2, change in line 15: just for better code formate, shorter and cleaner. lind of personal preference.

File: getDataFile/index.js (hunk 2 above)
Comment (mcqueen1987) [reply getDataFile/index.js]: the reason I change it is because the data in TT is very large, we could/should not get and save all data at once.
The best solutino is get like 1000 cases and save it, then get another 1000 cases. but in the previous framework could not support me to do that. so it's better to change the getDataFile to support get then save once or multiple times.
